src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -262,7 +262,10 @@ private async Task SignalEntityAsyncInternal(TaskHubClient client, string hubNam
                 throw new ArgumentNullException(nameof(operationName));
             }
 
-            this.config.ThrowIfFunctionDoesNotExist(entityId.EntityName, FunctionType.Entity);
+            if (client == this.client)
```

Comment:
I'm not convinced this block is ever true. That being said, it matches our logic for in other places. Do you mind just testing by using a TaskHub that matches the application's taskhub?

Reply:
I can change this to .Equals() to cover the case where the user specifies a TaskHub that matches the applications TaskHub.

Reply:
`==` and `Equals()` are equivalent assuming we haven't added our own implementation of `Equals()`, but I am fine with merging the current state.